pytorch_lightning/callbacks/early_stopping.py

```diff
@@ -160,15 +173,48 @@ def _run_early_stopping_check(self, trainer):
         # when in dev debugging
         trainer.dev_debugger.track_early_stopping_history(self, current)
 
-        if self.monitor_op(current - self.min_delta, self.best_score):
+        should_stop, reason = self._evalute_stopping_criteria(current)
+        if should_stop:
+            self.stopped_epoch = trainer.current_epoch
+        if reason:
+            log.info(f"[{trainer.global_rank}] {reason}")
```

Comment:
If we log on rank zero only, and the user has `sync_dist=False` for logging, then we might not see the reason being logged because it could be rank > 0 that decided to stop.
If we log on all ranks and the user has `sync_dist=True` for logging, we will show the same message N times.

Reply:
Good idea !